records/generator.go

```diff
@@ -303,22 +308,43 @@ func (rg *RecordGenerator) masterRecord(domain string, masters []string, leader
 	rg.insertRR(udp, host, "SRV")
```

Comment:
I'd love to see this method have no side effects. It would make it easier to reason about and test. We'd isolate the data generation from the mutation effect.
```go
records := rg.masterRecords(tc.domain, tc.masters...) // referentially transparent: same inputs, same outputs, no side r
if err := rg.insert(records...); err != nil {
   // handle error
}
```